ocaml/testsuite/tests/typing-modes/module.ml

```diff
@@ -0,0 +1,88 @@
+(* TEST
+   expect;
+*)
+
+(* This file tests that modules are sound wrt modes. *)
+
+module type S = sig val x : string end
+
+module type Empty = sig end
+
+module M = struct
+    let x = "string"
+end
+[%%expect{|
+module type S = sig val x : string end
+module type Empty = sig end
+module M : sig val x : string end
+|}]
+
+(* Closing over modules affects closure's modes *)
+let (foo @ portable) () =
+    let _ = (module M : S) in
+    ()
+(* CR zqian: This should fail *)
+[%%expect{|
+val foo : unit -> unit = <fun>
+|}]
+
+(* File-level modules are looked up differently and need to be tested
+separately. *)
+let (foo @ portable) () =
+    let _ = (module List : Empty) in
```

Comment:
There's a tiny chance this will be interpreted like `unit` and mode-cross. (It would actually be sound!) So maybe use a module type other than `Empty`.